decryptor/postgresql/pg_decryptor.go

```diff
@@ -408,6 +408,41 @@ func (proxy *PgProxy) sendClientError(msg string, logger *log.Entry) error {
 	return nil
 }
 
+// stopClientThread sends a signal to a client thread to stop. Returns error in
+// case of an error or timeout. Is used to stop and reload client with TLS
+func (proxy *PgProxy) stopClientThread(logger *log.Entry) error {
+	proxy.stopClient = true
+	// stop reading from client in goroutine
+	if err := proxy.clientConnection.SetDeadline(time.Now()); err != nil {
+		logger.
+			WithError(err).
+			WithField(logging.FieldKeyEventCode, logging.EventCodeErrorDecryptorCantSetDeadlineToClientConnection).
+			Errorln("Can't set deadline")
+		return err
+	}
+
+	select {
+	case <-proxy.ClientStopResponse:
+	case <-time.NewTimer(ClientStopTimeout).C:
+		logger.
+			// TODO: which event code
+			WithField(logging.FieldKeyEventCode, logging.EventCodeErrorDecryptorCantInitializeTLS).
+			Errorln("Can't stop background goroutine")
+		return errors.New("can't stop background goroutine")
+	}
+
+	// TODO: why twice?
```

Comment:
actually I don't remember) maybe to be sure)
or you can remove it and run several tests to check is it break something or not.

Reply:
Ohh, I guess this resets the timeout, because it uses `time.Time{}` instead of `time.Now()`